brave/brave-core
# fix(wallet): Bitcoin Local Storage Filters

## Description 
- Fixes a bug where collapsing a `Bitcoin Account` on the `Portfolio` screen would collapse all other `Bitcoin` accounts
- Fixes a bug where `Selecting` or `Deselecting` a `Bitcoin` account in the Portfolio `Filter and Display Settings`, would either `Select` or `Deselect` all `Bitcoin` accounts


Resolves <https://github.com/brave/brave-browser/issues/35425>
Resolves <https://github.com/brave/brave-browser/issues/35499>



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

### Testing Portfolio Account Collapsing:

   1. Create multiple `Bitcoin Testnet Accounts` and fund them with a `Bitcoin Testnet Faucet`
   2. Go to the `Portfolio` tab and sort by `Account`
   3. Collapse one of your `Bitcoin Accounts`
   4. It should not `Collapse` all other `Bitcoin Accounts`

Before:

https://github.com/brave/brave-core/assets/40611140/97296f29-1591-4773-b75f-d2b7f947a18b

After:

https://github.com/brave/brave-core/assets/40611140/3a7810a7-9453-4ff7-984c-4d4dafd32ff3

### Testing Portfolio Account Filtering:

   1. Create multiple `Bitcoin Testnet Accounts`
   2. Go to the Portfolio `Filter and Display Settings` and `Select` or `Deselect` one of the `Bitcoin` accounts
   3. It should not `Select` or `Deselect` all other `Bitcoin` accounts 

Before:

https://github.com/brave/brave-core/assets/40611140/8eff0a1b-91fd-4e39-a97e-f1c7c6c8ad76

After:

https://github.com/brave/brave-core/assets/40611140/06d503c5-9c5f-42bc-8863-3b3c9583bdf8
